app/components/custom_status/custom_status_expiry.tsx

```diff
@@ -0,0 +1,157 @@
+    const styles = createStyleSheet(theme);
+    const militaryTime = useSelector((state: GlobalState) => getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, 'use_military_time'));
+    const currentMomentTime = getCurrentMomentForTimezone(timezone);
+
+    const expiryMomentTime = timezone ? moment(time).tz(timezone) : moment(time);
+
+    const plusSixDaysEndTime = currentMomentTime.clone().add(6, 'days').endOf('day');
+    const tomorrowEndTime = currentMomentTime.clone().add(1, 'day').endOf('day');
+    const todayEndTime = currentMomentTime.clone().endOf('day');
+    const isCurrentYear = currentMomentTime.get('y') === expiryMomentTime.get('y');
+
+    let format = '';
+    let renderDayOrDate;
+
+    if (showToday && expiryMomentTime.isSameOrBefore(todayEndTime)) {
+        renderDayOrDate = (
+            <Text>
+                <FormattedText
+                    id='custom_status.expiry_time.today'
+                    defaultMessage='Today'
+                />
+                {' '}
+                <FormattedText
+                    id='custom_status.expiry.at'
+                    defaultMessage='at'
+                />
+                {' '}
+            </Text>
+        );
+    } else if (expiryMomentTime.isSame(todayEndTime)) {
+        renderDayOrDate = (
+            <FormattedText
+                id='custom_status.expiry_time.today'
+                defaultMessage='Today'
+            />
+        );
+    } else if (expiryMomentTime.isAfter(todayEndTime) && expiryMomentTime.isSameOrBefore(tomorrowEndTime)) {
+        renderDayOrDate = (
+            <Text>
+                <FormattedText
+                    id='custom_status.expiry_time.tomorrow'
+                    defaultMessage='Tomorrow'
+                />
+                {' '}
+                <FormattedText
+                    id='custom_status.expiry.at'
+                    defaultMessage='at'
+                />
+                {' '}
+            </Text>
+        );
+    }
+
+    const useDay = expiryMomentTime.isBetween(tomorrowEndTime, plusSixDaysEndTime);
+    const useDate = expiryMomentTime.isAfter(plusSixDaysEndTime) && isCurrentYear;
+
+    if (useDay) {
+        format = 'dddd';
+    } else if (useDate) {
+        format = 'MMM DD';
+    } else if (!isCurrentYear) {
+        format = 'MMM DD, YYYY';
+    }
+
+    if (expiryMomentTime.isAfter(tomorrowEndTime)) {
+        renderDayOrDate = (
+            <FormattedDate
+                format={format}
+                timezone={timezone}
+                value={expiryMomentTime.toDate()}
+            />
+        );
+
+        renderDayOrDate = showTimeCompulsory ? (
+            <Text>
+                {renderDayOrDate}
+                {' '}
+                <FormattedText
+                    id='custom_status.expiry.at'
+                    defaultMessage='at'
+                />
+                {' '}
+            </Text>
+        ) : renderDayOrDate;
+    }
```

Comment:
this block should be sorted to the other `if-else` blocks above ... simply define the format before running the conditions (in line 45) and you are good to go.